### PR TITLE
Health and metrics

### PR DESCRIPTION
**Pull request by gkertaseferi**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,355 @@
...
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.json.JsonObject;
+import javax.inject.Inject;
+```
+
+3. Now we need to add the required REST API calls to the class. We add one for each microservice. We are also using CDI to inject the class Proxy which is responsible for connecting to the back-end:
```
Comment (jamiecoleman92) [inline README.md]: This is correct except when I go onto branch (Module 5) I see this code already exists when I do not think it should as the user should be inserting this code on their own?

File: README.md [hunk 2]
```diff
@@ -1,4 +1,355 @@
...
+
+You can use the MicroProfile metrics API to add metrics to your applications. The MicroProfile metrics API is similar to the Dropwizard metrics API.
+
+When you use the MicroProfile metrics API, you can perform the following functions.
+1. Create and register metrics by using Contexts and Dependency Injection (CDI).
+2. Add metadata to describe metrics.
+3. Access useful JVM/server base metrics.
+4. Access the registries that contain all registered metrics and metadata.
+
+Some basic files have been provided for you such as a basic pom file for building our application with maven, a License file, some basic Open Liberty configuration in the form of xml and the directory structure required for this microservice.
```
Comment (jamiecoleman92) [inline README.md]: This has already been mentioned above!

File: README.md [hunk 3]
```diff
@@ -1,4 +1,355 @@
...
+    .build();
+      
+      // Stop timing
+      context.close();
+      return value;
+    }
+}
+```
+
+4. Next we need to update the `LibertyRestEnpoint.java` which is located in the same folder as the `JaxrsManager.java` to look like the example below adding the @ApplicationScoped annotation which creates the object once for the duration of the application. The @Timed annotation denotes a timer, which tracks duration of the annotated object. We add these annotations so we can register the metrics for the endpoints in the file.
```
Comment (jamiecoleman92) [inline README.md]: This files does not exist! LibertyRestEnpoint.java

File: README.md [hunk 4]
```diff
@@ -1,4 +1,355 @@
...
+10. When we go to `http://localhost:9090/metrics` we see JVM information. If we want to display metrics for our application we need to run our application first and then go to `http://localhost:9090/metrics/application`.
+
+
+### Application example that uses health and metrics endpoint of the microservice
+
+1. Navigate into the following directory `microservice-webapp/src/app/app.component.ts` 
+
+2. First declare the health and metrics endpoints for the microservice like this: 
+```
+healthOneEndpoint = "http://localhost:9080/LibertyGateway-1.0/rest/health1";
```
Comment (jamiecoleman92) [inline README.md]: This code already exists!

File: README.md [hunk 5]
```diff
@@ -1,4 +1,355 @@
...
+    .subscribe(data => {
+      this.metricsOne = data;
+      this.metricsOne = this.metricsOne.replace(/(base:)+/g, '\nbase:').replace(/(application:)+/g, '\napplication:');
+    });
+  }
+```
+
+5. Navigate into the following directory `microservice-webapp/src/app/app.component.html` 
+
+In the app.component.html you can write you own user interface, here is an example:
```
Comment (jamiecoleman92) [inline README.md]: This code already exists!

File: README.md (hunk 3 above)
Comment (gkertaseferi) [reply README.md]: This is the C:\work\tutorial-microprofile\microservice-one\src\main\java\application\rest\LibertyRestEndpoint.java not the one from the microservice-gateway.

File: README.md (hunk 1 above)
Comment (gkertaseferi) [reply README.md]: done :)

File: README.md (hunk 2 above)
Comment (gkertaseferi) [reply README.md]: done

File: README.md (hunk 4 above)
Comment (gkertaseferi) [reply README.md]: done :D

**Pull request merged**